### PR TITLE
Feature/ac/gsm

### PR DESCRIPTION
**Pull request by alyciacan**
# Description
* Mob-coded addition of MemeContext, using MemeContext in IndividualPainting component.

## Type of change

- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)
- [ ] Refactor

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] I tested my changes in the browser
- [ ] I tested with Mocha/Chai

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new bugs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Is your code D.R.Y?
- [x] Does it follow SRP?

# Questions:
Are there any questions or concerns you want addressed in a code review?
* None
# Next Steps:
* Cypress testing
* Get MyGallery linked up and rendering memes
* Styling for memes

### CODE REVIEW

**Review by Mlberry0205 — APPROVED** [review]
Beautiful work on this, team!

File: src/components/individualPainting/IndividualPainting.jsx [hunk 1]
```diff
@@ -22,7 +24,9 @@ const IndividualPainting = ({selectedId, memeTitle}) => {
     }
     findPainting()
   }, []);
+
   const { id, title, description, img, artist, year } = paintingObject;
+  const { memes, setMemes } = useContext(MemeContext);
```
Comment (letakeane) [inline src/components/individualPainting/IndividualPainting.jsx:29]: Nice job bringing in context!

File: src/components/myGallery/MyGallery.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import './MyGallery.css';
+import ArtCard from '../artCard/ArtCard';
+import { useContext } from 'react';
+import { MemeContext } from '../../contexts/MemeContext';
+
+const MyGallery = (arrayOfFavs) => {
+    const {memes} = useContext(memeContext)
+    const myFavs = arrayOfFavs.map(fav => {
```
Comment (letakeane) [inline src/components/myGallery/MyGallery.jsx:8]: Naming can be a bit tighter here - maybe something like "favoritedArtCards"

File: src/components/myGallery/MyGallery.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import './MyGallery.css';
+import ArtCard from '../artCard/ArtCard';
+import { useContext } from 'react';
+import { MemeContext } from '../../contexts/MemeContext';
+
+const MyGallery = (arrayOfFavs) => {
+    const {memes} = useContext(memeContext)
```
Comment (letakeane) [inline src/components/myGallery/MyGallery.jsx:7]: It doesn't look like `memes` is being used at all in this file. You may not need to bring in Context here?

File: src/contexts/MemeContext.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { createContext, useState } from 'react';
+
+export const MemeContext = createContext({
+ memes: null,
+ setMemes: () => null
+});
+
+export const MemeProvider = ({ children }) => {
```
Comment (letakeane) [inline src/contexts/MemeContext.js:8]: Interesting - I haven't personally run into creating a MemeProvider component this way!

File: src/components/myGallery/MyGallery.jsx (hunk 3 above)
Comment (alyciacan) [reply src/components/myGallery/MyGallery.jsx:7]: We hadn't built out MyGallery yet, we did end up using it eventually :)

**Pull request merged**